src/app/public/modules/text-editor/toolbar/text-editor-toolbar.component.ts

```diff
@@ -0,0 +1,217 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnInit,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  SkyColorpickerOutput,
+  SkyColorpickerMessage,
+  SkyColorpickerMessageType
+} from '@skyux/colorpicker';
+
+import {
+  SkyModalCloseArgs,
+  SkyModalService
+} from '@skyux/modals';
+
+import {
+  SkyDropdownMessage,
+  SkyDropdownMessageType
+} from '@skyux/popovers';
+
+import {
+  FONT_LIST_DEFAULTS
+} from '../defaults/font-list-defaults';
+
+import {
+  STYLE_STATE_DEFAULTS
+} from '../defaults/style-state-defaults';
+
+import {
+  SkyTextEditorService
+} from '../services/text-editor.service';
+
+import {
+  SkyTextEditorFont
+} from '../types/font-state';
+
+import {
+  SkyTextEditorStyleState
+} from '../types/style-state';
+
+import {
+  SkyTextEditorToolbarActions
+} from '../types/toolbar-action';
+
+import {
+  SkyTextEditorUrlModalComponent
+} from '../url-modal/text-editor-url-modal.component';
+
+import {
+  UrlTarget
+} from '../url-modal/text-editor-url-target';
+
+import {
+  SkyUrlModalContext
+} from '../url-modal/text-editor-url-modal-context';
+
+/**
+ * @internal
+ */
+@Component({
+  selector: 'sky-text-editor-toolbar',
+  templateUrl: './text-editor-toolbar.component.html',
+  styleUrls: ['./text-editor-toolbar.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None
+})
+export class SkyTextEditorToolbarComponent implements OnInit {
+
+  @Input()
```

Comment:
Do the public inputs in this file need JSDocs comments?

Reply:
Nope, this is an internal component.